pkg/hub/managedcluster/manifests/managedcluster-registration-role.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .ManagedClusterName }}:managed-cluster-registration
+  namespace: {{ .ManagedClusterName }}
+rules:
+# Allow spoke registration agent to create/get/update coordination.k8s.io/lease
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["create", "get", "update"]
```

Comment:
I expected the get and update to be restricted based on name.  Otherwise any spoke can be aware of any other spoke's name and status.

Reply:
> I expected the get and update to be restricted based on name. Otherwise any spoke can be aware of any other spoke's name and status.

keep in mind, this means that there is no ability for the spoke to create.

Reply:
I also expect restriction based on name

Reply:
done